aws-http-client-java/src/main/java/software/amazon/awssdk/http/java/JavaHttpClient.java

```diff
@@ -0,0 +1,140 @@
+import software.amazon.awssdk.http.SdkHttpFullResponse;
+import software.amazon.awssdk.http.SdkRequestContext;
+import software.amazon.awssdk.utils.AttributeMap;
+import software.amazon.awssdk.utils.IoUtils;
+
+public final class JavaHttpClient implements SdkHttpClient {
+
+    private final AttributeMap options;
+
+    public JavaHttpClient(AttributeMap options) {
+        this.options = options;
+    }
+
+    @Override
+    public AbortableCallable<SdkHttpFullResponse> prepareRequest(SdkHttpFullRequest request, SdkRequestContext requestContext) {
+        final HttpURLConnection connection = createAndConfigureConnection(request);
+
+        return new AbortableCallable<SdkHttpFullResponse>() {
+
+            @Override
+            public SdkHttpFullResponse call() throws Exception {
+                connection.connect();
+
+                if (request.getContent() != null) {
+                    IoUtils.copy(request.getContent(), connection.getOutputStream());
+                }
+
+                int responseCode = connection.getResponseCode();
+                return SdkHttpFullResponse.builder()
+                                          .statusCode(responseCode)
+                                          .statusText(connection.getResponseMessage())
+                                          .content(connection.getInputStream())
+                                          .headers(extractHeaders(connection))
+                                          .build();
+            }
+
+
+            private Map<String, List<String>> extractHeaders(HttpURLConnection response) {
+                return response.getHeaderFields().entrySet().stream().filter(e -> e.getKey() != null).collect(toMap());
+            }
+
+            @Override
+            public void abort() {
+                if (connection != null) {
+                    connection.disconnect();
+                }
+            }
+        };
+    }
+
+    private HttpURLConnection createAndConfigureConnection(SdkHttpFullRequest request) {
+        HttpURLConnection connection = invokeSafely(() -> (HttpURLConnection) createRequest(request).toURL().openConnection());
+        request.getHeaders().forEach((key, values) -> values.forEach(value -> connection.setRequestProperty(key, value)));
+        invokeSafely(() -> connection.setRequestMethod(request.getHttpMethod().name()));
+        if (request.getContent() != null) {
+            connection.setDoOutput(true);
+        }
+
+        connection.setConnectTimeout(saturatedCast(options.get(CONNECTION_TIMEOUT).toMillis()));
+        connection.setReadTimeout(saturatedCast(options.get(SOCKET_TIMEOUT).toMillis()));
+
+        return connection;
+    }
+
+    @Override
+    public <T> Optional<T> getConfigurationValue(SdkHttpConfigurationOption<T> key) {
+        return Optional.ofNullable(options.get(key));
+    }
+
+    private URI createRequest(SdkHttpFullRequest request) {
+        String params = request.getParameters().entrySet().stream()
```

Comment:
Moved so that it's right after where it's called, and put all the `public` methods together.